addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/wso2iotsHandler.java

```diff
@@ -0,0 +1,390 @@
+                        updateChannel(wso2iotsBindingConstants.MOTION, iotsResponseMotion);
+                        updateChannel(wso2iotsBindingConstants.HUMIDITY, iotsResponseHumidity);
+
+                        // for (Channel channel : getThing().getChannels()) {
+                        // updateChannel(channel.getUID().getId(), iotsResponse);
+                        // }
+                    } catch (Exception e) {
+                        logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                    }
+                }
+            };
+
+            wso2iotsConfiguration config = getConfigAs(wso2iotsConfiguration.class);
+            int delay = (config.refresh != null) ? config.refresh.intValue() : DEFAULT_REFRESH_PERIOD;
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.MINUTES);
+        }
+    }
+
+    /**
+     * Update the channel from the last building monitor data retrieved
+     *
+     * @param channelId the id identifying the channel to be updated
+     */
+    private void updateChannel(String channelId, BigDecimal iotsResponse) {
+        if (isLinked(channelId)) {
+            /*
+             * Object value;
+             * try {
+             * value = getValue(channelId, iotsResponse);
+             * } catch (Exception e) {
+             * logger.debug("Station doesn't provide {} measurement", channelId.toUpperCase());
+             * return;
+             * }
+             */
+
+            State state = null;
+            if (iotsResponse == null) {
+                state = UnDefType.UNDEF;
+            } else if (iotsResponse instanceof BigDecimal) {
+                state = new DecimalType(iotsResponse);
+
+            } else {
+                logger.warn("Update channel {}: Unsupported value type {}", channelId,
+                        iotsResponse.getClass().getSimpleName());
+            }
+            logger.debug("Update channel {} with state {} ({})", channelId, (state == null) ? "null" : state.toString(),
+                    (iotsResponse == null) ? "null" : iotsResponse.getClass().getSimpleName());
+
+            // Update the channel
+            if (state != null) {
+                updateState(channelId, state);
+            }
+        }
+    }
+
+    private BigDecimal updateBuildingMonitorData(String sensorType) throws Exception {
+
+        BigDecimal iotsResponse = null;
+        boolean resultok = false;
+        String errormsg = null;
+        String response1 = null;
+        wso2iotsConfiguration config = getConfigAs(wso2iotsConfiguration.class);
+        String Access_Token = config.apikey;
+        String id = config.deviceId;
+        // String Access_Token = "bfe52f8a-5f34-382b-802c-93e52468564f";
+        // String id = "2940205";
+        long time = System.currentTimeMillis() / 1000;
+        String fromTime = String.valueOf(time - 200);
```

Comment:
why 200? should this be a configuration parameter? otherwise, please use a constant